src/gtk/toplevel.cpp

```diff
@@ -1602,8 +1600,9 @@ bool wxTopLevelWindowGTK::SetTransparent(wxByte alpha)
         }
         return true;
     }
-#endif // GTK_CHECK_VERSION(2,12,0)
-#ifndef __WXGTK3__
+
+    return false;
```

Comment:
This looks wrong. The 2.12 check above won't fail on GTK3, but it could on GTK2, and in that case we want to continue with the code below. I think this part should stay as it was.

Reply:
Yes, this is indeed wrong, thanks. But this code is so hairy that I'd like to simplify it a bit instead of leaving the current unclear (even if correct) chain of checks. What about the new version separating GTK+ 2 and 3 versions, at the (IMO perfectly acceptable) cost of duplicating `gtk_window_set_opacity()` call?